plugins/inputs/kafka_consumer_legacy/kafka_consumer_legacy.go

```diff
@@ -91,6 +91,7 @@ func (k *Kafka) Start(acc telegraf.Accumulator) error {
 	defer k.Unlock()
 	var consumerErr error
 
+	k.done = make(chan struct{})
```

Comment:
Is this related to config api? It seems like moving initialization of k.done up a few lines shouldn't matter.